### PR TITLE
Added request line size check

### PR DESCRIPTION
**Pull request by qrilka**

### CODE REVIEW

File: src/mochiweb_http.erl [hunk 1]
```diff
@@ -115,6 +122,22 @@ headers(Socket, Request, Headers, Body, HeaderCount) ->
         exit(normal)
     end.
 
+to_list(A) when is_atom(A) ->
+    atom_to_list(A);
+to_list(L) when is_list(L) ->
+    L.
+
+with_valid_line_size(RequestLine, Socket, Request, Body, Fun) ->
```
Comment (dmzmk) [inline src/mochiweb_http.erl]: `Body` is unused here

File: src/mochiweb_http.erl [hunk 2]
```diff
@@ -98,8 +101,12 @@ headers(Socket, Request, Headers, Body, HeaderCount) ->
...
             ?MODULE:after_response(Body, Req);
         {Protocol, _, {http_header, _, Name, _, Value}} when Protocol == http orelse Protocol == ssl ->
-            headers(Socket, Request, [{Name, Value} | Headers], Body,
-                    1 + HeaderCount);
+            with_valid_line_size([to_list(Name), $:, Value, "\r\n"], 
+                                 Socket, Request, Body,
+                                 fun() ->
+                                         headers(Socket, Request, [{Name, Value} | Headers], Body,
+                                                 1 + HeaderCount)
+                                 end);
```
Comment (dmzmk) [inline src/mochiweb_http.erl]: Why do we need such helper here? 
```erlang
{Protocol, _, {http_header, _, Name, _, Value}} when Protocol == http orelse Protocol == ssl ->
    case iolist_size(to_list(Name)) + iolist_size(Value) > ?REQUEST_LINE_LIMIT of
        true ->
            NewRequest = new_request(Socket, Request, []),
            NewRequest:respond({413, [], "Request line too large"}),
            mochiweb_socket:close(Socket),
            exit(normal);
        false ->
            headers(Socket, Request, [{Name, Value} | Headers], Body, 1 + HeaderCount)
    end;
```
IMHO this is more readable.

Comment (qrilka) [inline src/mochiweb_http.erl]: the idea was to use it not only for headers but later I've decided that there is not particular need to "support" requests with URLs larger than 8k :)
Will inline it

Comment (dmzmk) [inline src/mochiweb_http.erl]: Ah, ok :+1:

**Comment by qrilka** [comment]
Inlined

**Pull request merged**